Partial/PartialNode.php

```diff
@@ -54,6 +49,14 @@ public static function excludeEverythingNotInPath(ArrayNode $node, array $path =
         }
         self::nodeChildrenProperty()->setValue($node, $children);
 
+        if (!($nextNode instanceof ArrayNode)) {
+            if (!empty($path)) {
+                throw new ChildIsNotAnArrayNode($node, $nextNodeName);
+            }
+
+            return;
+        }
```

Comment:
Ah, but we want to throw exception if we reach a scalar node, but path indicates we still have some processing to do. Which we cannot do of course, because the node is scalar and there's nothing left to process.

Only if path is empty and node is scalar is a valid situation (in which case we simply return)